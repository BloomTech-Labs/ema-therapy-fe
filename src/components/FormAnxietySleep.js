import React from 'react';
import Slider from 'antd/es/slider';
// import 'antd/dist/antd.css';

const FormAnxietySleep = ({
  handleView,
  anxietyLevel,
  onAnxietySliderChange,
  handleChange,
  sleep,
}) => {
  return (
    <>
      <div className="header">
        <button
          type="button"
          className="back"
          onClick={() => handleView('activity-journal')}
        >
          &larr;
        </button>
        <p>How stressed are you this moment from 1 - 10?</p>
      </div>
      <div className="inputs-section">
        <div className="inputs">
          <p>Anxiety Level</p>
          <Slider
            value={anxietyLevel}
            onChange={onAnxietySliderChange}
            min={1}
            max={10}
          />
        </div>
        <div className="inputs">
          <label htmlFor="sleep">
            Hours of sleep:
            <input
              type="number"
              name="sleep"
              id="sleep"
              value={sleep}
              onChange={handleChange}
            />
          </label>
        </div>
      </div>
      <div className="footer">
        <button className="main-button" type="submit">
          Done
        </button>
      </div>
    </>
  );
};

export default FormAnxietySleep;
