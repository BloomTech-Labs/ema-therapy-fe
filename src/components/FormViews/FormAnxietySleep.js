import React from 'react';
import styled from 'styled-components';
import { Slider, Icon } from 'antd';
import PropTypes from 'prop-types';
import NextButton from './NextButton';

function FormAnxietySleep({
  sleep,
  handleView,
  anxietyLevel,
  onSleepSliderChange,
  onAnxietySliderChange,
}) {
  return (
    <>
      <div className="header">
        <Icon
          className="back-btn"
          type="left"
          data-testid="back"
          onClick={() => handleView('mood')}
        />
        <p>
          How stressed <br />
          are you?
        </p>
      </div>
      <div className="inputs-section">
        <div className="inputs">
          <Label>Anxiety Level</Label>
          <SliderWrapper>
            <span>0</span>
            <Slider
              value={anxietyLevel}
              onChange={onAnxietySliderChange}
              min={0}
              max={10}
              style={{ flexGrow: 1 }}
            />
            <span>10</span>
          </SliderWrapper>
          <Label>Hours of sleep (since last entry)</Label>
          <SliderWrapper>
            <span>0</span>
            <Slider
              value={sleep}
              onChange={onSleepSliderChange}
              min={0}
              max={10}
              style={{ flexGrow: 1 }}
            />
            <span>10+</span>
          </SliderWrapper>
        </div>
      </div>
      <div className="footer">
        <NextButton data-testid="next" onClick={() => handleView('activities')}>
          Next
        </NextButton>
      </div>
    </>
  );
}

FormAnxietySleep.propTypes = {
  handleView: PropTypes.func.isRequired,
  anxietyLevel: PropTypes.number.isRequired,
  onAnxietySliderChange: PropTypes.func.isRequired,
  onSleepSliderChange: PropTypes.func.isRequired,
  sleep: PropTypes.number.isRequired,
};

export default FormAnxietySleep;

const Label = styled.p`
  color: #00917a;
  font-size: 18px;
  padding-left: 20px;
  margin-bottom: 10px;
`;

const SliderWrapper = styled.div`
  display: flex;
  align-items: center;
  &:first-of-type {
    margin-bottom: 72px;
  }

  span {
    color: #00917a;
    font-size: 14px;
    margin-top: 5px;
    &:first-of-type {
      padding-right: 8px;
    }

    &:not(:first-of-type) {
      padding-left: 8px;
    }
  }
`;
