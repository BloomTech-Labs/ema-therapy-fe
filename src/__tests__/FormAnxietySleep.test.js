import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FormAnxietySleep from '../components/FormViews/FormAnxietySleep';
import 'jest-styled-components';

describe('FormAnxietySleep component', () => {
  test('renders correctly', () => {
    expect(
      render(
        <FormAnxietySleep
          handleView={() => {}}
          anxietyLevel={7}
          onAnxietySliderChange={() => {}}
          handleChange={() => {}}
          onSleepSliderChange={() => {}}
          handleSubmit={() => {}}
          isSubmitting
          sleep={5}
        />,
      ),
    ).toMatchSnapshot();
  });
  test('changes view to activity-journal on back button', () => {
    const handleViewMock = jest.fn();
    const { getByTestId } = render(
      <FormAnxietySleep
        handleView={handleViewMock}
        anxietyLevel={7}
        onAnxietySliderChange={() => {}}
        onSleepSliderChange={() => {}}
        handleChange={() => {}}
        handleSubmit={() => {}}
        isSubmitting
        sleep={5}
      />,
    );
    const backButton = getByTestId('back');
    fireEvent.click(backButton);
    expect(handleViewMock).toHaveBeenLastCalledWith('mood');
  });
});
test('onAnxietySliderChange gets called when slider is moved', () => {
  const antSlider = jest.fn();
  const { getAllByRole } = render(
    <FormAnxietySleep
      handleView={() => {}}
      anxietyLevel={7}
      onAnxietySliderChange={antSlider}
      handleChange={() => {}}
      onSleepSliderChange={() => {}}
      handleSubmit={() => {}}
      isSubmitting
      sleep={5}
    />,
  );
  const moveSlider = getAllByRole('slider');
  fireEvent.mouseDown(moveSlider[0]);
  expect(antSlider).toHaveBeenCalled();
});
test('sleep number changes when sleep slider moves', () => {
  const antSliderSleep = jest.fn();
  const { getAllByRole } = render(
    <FormAnxietySleep
      handleView={() => {}}
      anxietyLevel={7}
      onAnxietySliderChange={() => {}}
      onSleepSliderChange={antSliderSleep}
      handleSubmit={() => {}}
      isSubmitting
      sleep={5}
    />,
  );
  const moveSlider = getAllByRole('slider');
  fireEvent.mouseDown(moveSlider[1]);
  expect(antSliderSleep).toHaveBeenCalled();
});
