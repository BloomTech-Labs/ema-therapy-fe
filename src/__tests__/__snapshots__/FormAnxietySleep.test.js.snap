// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAnxietySleep component handleChange gets called when a number is entered into the sleep hours field 1`] = `
Array [
  <div
    className="header"
  >
    <button
      className="back"
      onClick={[Function]}
      type="button"
    >
      ←
    </button>
    <p>
      How stressed are you this moment from 1 - 10?
    </p>
  </div>,
  <div
    className="inputs-section"
  >
    <div
      className="inputs"
    >
      <p>
        Anxiety Level
      </p>
      <div
        className="ant-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="ant-slider-rail"
          style={Object {}}
        />
        <div
          className="ant-slider-track"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "width": "66.66666666666666%",
            }
          }
        />
        <div
          className="ant-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={10}
          aria-valuemin={1}
          aria-valuenow={7}
          className="ant-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="slider"
          style={
            Object {
              "left": "66.66666666666666%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={0}
        />
        <div
          className="ant-slider-mark"
        />
      </div>
    </div>
    <div
      className="inputs"
    >
      <label
        htmlFor="sleep"
      >
        Hours of sleep:
        <input
          id="sleep"
          name="sleep"
          onChange={[MockFunction]}
          type="number"
          value="testString"
        />
      </label>
    </div>
  </div>,
  <div
    className="footer"
  >
    <button
      className="main-button"
      type="submit"
    >
      Done
    </button>
  </div>,
]
`;

exports[`FormAnxietySleep component handleView gets called with correct string when clicking back button 1`] = `
Array [
  <div
    className="header"
  >
    <button
      className="back"
      onClick={[Function]}
      type="button"
    >
      ←
    </button>
    <p>
      How stressed are you this moment from 1 - 10?
    </p>
  </div>,
  <div
    className="inputs-section"
  >
    <div
      className="inputs"
    >
      <p>
        Anxiety Level
      </p>
      <div
        className="ant-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="ant-slider-rail"
          style={Object {}}
        />
        <div
          className="ant-slider-track"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "width": "66.66666666666666%",
            }
          }
        />
        <div
          className="ant-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={10}
          aria-valuemin={1}
          aria-valuenow={7}
          className="ant-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="slider"
          style={
            Object {
              "left": "66.66666666666666%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={0}
        />
        <div
          className="ant-slider-mark"
        />
      </div>
    </div>
    <div
      className="inputs"
    >
      <label
        htmlFor="sleep"
      >
        Hours of sleep:
        <input
          id="sleep"
          name="sleep"
          onChange={[Function]}
          type="number"
          value="testString"
        />
      </label>
    </div>
  </div>,
  <div
    className="footer"
  >
    <button
      className="main-button"
      type="submit"
    >
      Done
    </button>
  </div>,
]
`;

exports[`FormAnxietySleep component renders correctly 1`] = `
Array [
  <div
    className="header"
  >
    <button
      className="back"
      onClick={[Function]}
      type="button"
    >
      ←
    </button>
    <p>
      How stressed are you this moment from 1 - 10?
    </p>
  </div>,
  <div
    className="inputs-section"
  >
    <div
      className="inputs"
    >
      <p>
        Anxiety Level
      </p>
      <div
        className="ant-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="ant-slider-rail"
          style={Object {}}
        />
        <div
          className="ant-slider-track"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "width": "66.66666666666666%",
            }
          }
        />
        <div
          className="ant-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={10}
          aria-valuemin={1}
          aria-valuenow={7}
          className="ant-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="slider"
          style={
            Object {
              "left": "66.66666666666666%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={0}
        />
        <div
          className="ant-slider-mark"
        />
      </div>
    </div>
    <div
      className="inputs"
    >
      <label
        htmlFor="sleep"
      >
        Hours of sleep:
        <input
          id="sleep"
          name="sleep"
          onChange={[Function]}
          type="number"
          value="testString"
        />
      </label>
    </div>
  </div>,
  <div
    className="footer"
  >
    <button
      className="main-button"
      type="submit"
    >
      Done
    </button>
  </div>,
]
`;
